tests/devices/eiger/test_eiger.py

```diff
@@ -0,0 +1,254 @@
+import json
+
+import aiohttp
+import pytest
+from aiohttp import web
+from mock import MagicMock, Mock
+from mock.mock import create_autospec
+
+from tickit.devices.eiger.eiger import EigerDevice, EigerRESTAdapter
+from tickit.devices.eiger.eiger_settings import EigerSettings
+from tickit.devices.eiger.eiger_status import EigerStatus, State
+
+# # # # # EigerDevice Tests # # # # #
+
+
+@pytest.fixture
+def eiger() -> EigerDevice:
+    return EigerDevice()
+
+
+def test_eiger_constructor():
+    EigerDevice()
+
+
+@pytest.mark.asyncio
+async def test_eiger_initialize(eiger: EigerDevice):
+    await eiger.initialize()
+
+    assert State.IDLE.value == eiger.get_state()["value"]
+
+
+@pytest.mark.asyncio
+async def test_eiger_arm(eiger: EigerDevice):
+    await eiger.arm()
+
+    assert State.READY.value == eiger.get_state()["value"]
+
+
+@pytest.mark.asyncio
+async def test_eiger_disarm(eiger: EigerDevice):
+    await eiger.disarm()
+
+    assert State.IDLE.value == eiger.get_state()["value"]
+
+
+@pytest.mark.asyncio
+async def test_eiger_trigger_ints_and_ready(eiger: EigerDevice):
+
+    eiger._set_state(State.READY)
+    eiger.settings.trigger_mode = "ints"
+
+    message = await eiger.trigger()
+
+    assert State.ACQUIRE.value == eiger.get_state()["value"]
+    assert "Aquiring Data from Eiger..." == message
+
+
+@pytest.mark.asyncio
+async def test_eiger_trigger_not_ints_and_ready(eiger: EigerDevice):
+
+    eiger._set_state(State.READY)
+    # Should be 'exts' by default but set just in case
+    eiger.settings.trigger_mode = "exts"
```

Comment:
We shouldn't have "just in case" sets in tests, omitting it will effectively serve as a regression test against this behavior being default